### DISCUSSION TITLE
INAV 6.0 AHRS Beta Test Feedback

### DISCUSSION THREAD

**Discussion by b14ckyy** [iNavFlight/inav#8469] [General]
I'd like to open this discussion thread to collect feedback and discuss the AHRS Update https://github.com/iNavFlight/inav/pull/8403
@shota3527 is currently preparing the code into a "final" commit so we can release a beta 6.0 Version for public testing. Some notes and advises for the beta test will follow.

**Comment by b14ckyy** [comment]
For the Release notes @DzikuVx: 

### **Updated AHRS (Attitude & Heading Reference System)**
-----------------------------
INAV 6.0 includes a big Update to its AHRS for Attitude Estimation, to make sure INAV always knows it's correct attitude relative to the ground. This should once and for all fix the issue, known as "Horizon Drift" and makes any navigation mode as well as self-leveling modes like ANGLE fully reliable and allows much more precise GPS-Navigation. This affects Fixed Wing as well as Multirotor. 

The best results are given if the craft is equipped with GPS and in the case of Multirotor also with a compass. But also for pure LOS Crafts that have no GPS on board, the AHI stability is noticeably improved in most situations. To work best with non-GPS Fixed Wings, the Reference Airspeed has to be set according the average cruise speed of the craft. 

---------------------------
### Beta only: When Updating from INAV 5.1 or older, apply the following settings after loading back your previous DIFF (Any Fixed Wing or Copter): 

```
set imu_acc_ignore_rate = 15
set imu_acc_ignore_slope = 10
set imu_dcm_kp = 2000
set imu_dcm_ki = 50
set imu_dcm_kp_mag = 2000
set imu_dcm_ki_mag = 50
```

Fixed Wing only: 
`set imu_inertia_comp_method = ADAPTIVE
`
Copter only: 
`set imu_inertia_comp_method = VELNED
`
**For Fixed Wing Craft with no GPS:** 
`set fw_reference_airspeed = `
_(set this in cm/s. Set this to airspeed at which PIDs were tuned. Usually should be set to cruise airspeed. Also used for coordinated turn calculation if airspeed sensor is not present.)_

If you notice any problems, please head to the Discussion Thread here and post your feedback, if possible with DVR and Blackbox log: https://github.com/iNavFlight/inav/discussions/8469

**Reply by JulioCesarMatias** [reply]
> New AHRS

`New AHRS`  We don't have a new AHRS here, just two things: Limit max omega drift (aka gyro) and correct omega drift based on GPS Velocity NED.

`complete rework of the AHRS` this was not done either...

**Reply by b14ckyy** [reply]
@JulioCesarMatias I think you are right. I'll update the wording.

**Comment by stronnag** [comment]
Just FYI, the [seyrsnys.myzen.co.uk INAV Next](http://seyrsnys.myzen.co.uk/inav_dev/) repo builds how include #8403

**Reply by b14ckyy** [reply]
Nice! I think Pawel also wants to make an official Beta release that will be accessible in Configurator. Just not sure if he will also publish a configurator beta or link http://seyrsnys.myzen.co.uk/inav-configurator-next/ 

We just want to make sure that we stay on the same tested and stable build.

**Comment by DzikuVx** [comment]
https://github.com/iNavFlight/inav/wiki/6.0.0-Release-Notes

**Reply by b14ckyy** [reply]
Not sure if you will build the Preview from Master or from Shotas commit but can you please also include https://github.com/iNavFlight/inav/pull/8434 when you do the builds?

**Comment by P-I-Engineer** [comment]
flew for a while today, sat in a loiter, both directions. flew around in acro for a while.

horizon performed great, zero complaints. good work.

**Comment by P-I-Engineer** [comment]
dragon 2 and t-1 flying today, ripping some turns, horizon stayed level.

**Comment by jreise-d** [comment]
Tested several flights over 2 days on a MPX Twinstar with F411wing and am impressed, no visible horizon drift! Until now I had 2.6.1 on all my inav models, as this was the last working version. 3.0, 4.1 and 5.1 did not work for me safely because of the strong drift. Now I can finally update and use newer flight controllers. Many thanks!

**Comment by 0crap** [comment]
Tested this weekend with 4 packs on the AR.Wing 900
Works great!
I do notice, when I do fast tight circles, the horizon gets a bit floaty and shows a slight offset.
But when I suddenly stop the circle and fly straight again, the horizon immediately is spot on.

Great improvement!

**Reply by b14ckyy** [reply]
yes the g-force copmpensation has limits. The GPS takes a moment to recognize very fast directional changes like sharp turns so the compensation can be slightly off and if your plane turns with more than 80°/s, then there is still a classical ignore rate coming into place, that disables all accellerometer readings. So there are extreme edge cases where the AHI can still drift but this quickly recovers as soon as the plane flies "normal" again.

**Comment by kasatka60** [comment]
Tested on Z-84 and Mini Talon. Everything works perfectly. The horizon did not drift.

**Comment by MikeZakhar** [comment]
Tested today on airplane Heewing T1 (750mm wingspan, dual motors), controller matek F411-WTE, battery Li-ion 3S1P of 21700. Seems found a bug: In the flight summary some data are incorrect
<img width="711" alt="Снимок экрана 2023-01-07 в 19 26 28" src="https://user-images.githubusercontent.com/122180623/211170194-5c855c02-0528-4008-a462-2ee15cbb0dc4.png">
<img width="710" alt="Снимок экрана 2023-01-07 в 19 28 24" src="https://user-images.githubusercontent.com/122180623/211170200-5d336ca7-f925-49f1-aa3e-7e107001231c.png">
Max speed, altitude, current, min battery voltage - incorrect. It seems that is not max/min values but values just after disarm or landing.
By the way horizon drift issue sims gone 👍, great work!
